pkg/instance/instance.go

```diff
@@ -959,6 +977,37 @@ func (i *Instance) SetPrometheusRemoteWriteExporter(endpoint string) error {
 	return nil
 }
 
+// TsharkCollectorEnabled returns true if the tshark collector is enabled
+func (i *Instance) TsharkCollectorEnabled() bool {
+	return i.tsharkCollectorConfig != nil
+}
+
+// EnableTsharkCollector enables the tshark collector for the instance
+// This function can only be called in the state 'Preparing' or 'Committed'
+func (i *Instance) EnableTsharkCollector(volumeSize, s3AccessKey, s3SecretKey, s3Region, s3BucketName, s3KeyPrefix string) error {
+	// volumeSize, s3AccessKey, s3SecretKey, s3Region, s3BucketName, s3KeyPrefix string
+	if err := i.validateStateForObsy("Tshark collector"); err != nil {
+		return err
+	}
+	if i.TsharkCollectorEnabled() {
+		return ErrTsharkCollectorAlreadyEnabled
+	}
+	// prefix can be empty, so it's not needed to check it
+	if volumeSize == "" || s3AccessKey == "" || s3SecretKey == "" || s3Region == "" || s3BucketName == "" {
+		return ErrTsharkCollectorConfigNotSet.WithParams(volumeSize, "REDACTED", "REDACTED", s3Region, s3BucketName)
```

Comment:
[non-blocking] if we never return the access and secret key we should just removed them from the error message. Then when we are debugging or testing we can just add a log statement. That would prevent accidental logging of these keys in the future if we forgot to revert the `REDACTED` by accident.